packages/react-query/src/__tests__/useSuspenseQuery.test.tsx

```diff
@@ -52,22 +54,42 @@ describe('useSuspenseQuery', () => {
       )
     }
 
-    const rendered = renderWithClient(
-      queryClient,
+    const rendered = await renderStream.render(
       <React.Suspense fallback="loading">
         <Page />
       </React.Suspense>,
+      {
+        wrapper: ({ children }) => <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+      }
     )
 
-    await waitFor(() => rendered.getByText('data: 1'))
+    {
+      const { snapshot, withinDOM } = await renderStream.takeRender()
+      // Page didn't render yet, so no call to `replaceSnapshot` yet
+      expect(snapshot).toBeUndefined() // I'd probably skip this assertion
+      withinDOM().getByText("loading")
+    }
+    {
+      const { snapshot, withinDOM } = await renderStream.takeRender()
+      expect(snapshot).toMatchObject({ data: 1, status: 'success' })
+      withinDOM().getByText("data: 1") // I'd probably skip this assertion
+    }
     fireEvent.click(rendered.getByLabelText('toggle'))
-
-    await waitFor(() => rendered.getByText('data: 2'))
-
-    expect(renders).toBe(4)
```

Comment:
no need to explicitly assert on four renders anymore - we called `takeRender` four times, and we assert that no more renders will happen at the end of the test